pkg/sorters/bogo.go

```diff
@@ -0,0 +1,29 @@
+package sorters
+
+import "sync"
+import "math/rand"
+
+func Bogo(arr []int, update chan<- int, mutex *sync.Mutex) {
+	n := len(arr)
+
+	for {
+		rand.Shuffle(n, func(i, j int) {
+			mutex.Lock()
+			arr[i], arr[j] = arr[j], arr[i]
+			mutex.Unlock()
+			update <- i
+			update <- j
+		})
+
+		sorted := true
+		for i := 0; i < n - 1; i++ {
+			if arr[i] > arr[i + 1] {
+				sorted = false
+			}
+		}
```

Comment:
Can you do an early break out of the loop if `sorted` is false. For example if the first two are not sorted we don't want to go through the rest of the elements. Instead it should immediately go back to shuffling.